flood_adapt/integrator/sfincs_adapter.py

```diff
@@ -486,46 +929,56 @@ def read_zs_points(self):
         gdf = gpd.GeoDataFrame(
             pt_df,
             geometry=gpd.points_from_xy(da.point_x.values, da.point_y.values),
-            crs=self.sf_model.crs,
+            crs=self._model.crs,
         )
         return df, gdf
 
-    def get_mask(self):
-        """Get mask with inactive cells from model."""
-        mask = self.sf_model.grid["msk"]
-        return mask
+    ### OUTPUT HELPERS ###
+    def _write_floodmap_geotiff(self):
+        results_path = self._get_result_path()
 
-    def get_bedlevel(self):
-        """Get bed level from model."""
-        self.sf_model.read_results()
-        zb = self.sf_model.results["zb"]
-        return zb
+        for sim_path in self._get_simulation_paths():
+            # read SFINCS model
+            with SfincsAdapter(model_root=sim_path, site=self._site) as model:
+                # dem file for high resolution flood depth map
+                demfile = Database().static_path.joinpath(
+                    "dem", self._site.attrs.dem.filename
+                )
 
-    def get_model_boundary(self) -> gpd.GeoDataFrame:
-        """Get bounding box from model."""
-        return self.sf_model.region
+                # read max. water level
+                zsmax = model._get_zsmax()
 
-    def get_model_grid(self) -> QuadtreeGrid:
-        """Get grid from model.
+                # writing the geotiff to the scenario results folder
+                model._write_geotiff(
+                    zsmax,
+                    demfile=demfile,
+                    floodmap_fn=results_path.joinpath(
+                        f"FloodMap_{self._scenario.attrs.name}.tif"
+                    ),
+                )
 
-        Returns
-        -------
-        QuadtreeGrid
-            QuadtreeGrid with the model grid
-        """
-        return self.sf_model.quadtree
+    def _write_water_level_map(self):
+        """Read simulation results from SFINCS and saves a netcdf with the maximum water levels."""
+        # read SFINCS model
+        sim_paths = self._get_simulation_paths()
+        results_path = self._get_result_path()
+
+        # TODO investigate: why only read one model?
+        with SfincsAdapter(model_root=sim_paths[0], site=self._site) as model:
+            zsmax = model.read_zsmax()
+            zsmax.to_netcdf(results_path.joinpath("max_water_level_map.nc"))
 
-    def write_geotiff(self, zsmax, demfile: Path, floodmap_fn: Path):
+    def _write_geotiff(self, zsmax, demfile: Path, floodmap_fn: Path):
```

Comment:
Are we using this anywhere? I think this can go. This might have been me trying to debug and I wanted to output the water levels as well